packages/entity/src/EntityMutationTrigger.ts

```diff
@@ -2,6 +2,19 @@ import { EntityQueryContext } from './EntityQueryContext';
 import ReadonlyEntity from './ReadonlyEntity';
 import ViewerContext from './ViewerContext';
 
+export interface EntityMutationValidatorConfiguration<
+  TFields,
+  TID,
+  TViewerContext extends ViewerContext,
+  TEntity extends ReadonlyEntity<TFields, TID, TViewerContext, TSelectedFields>,
+  TSelectedFields extends keyof TFields = keyof TFields
+> {
+  beforeCreate?: EntityMutationTrigger<TFields, TID, TViewerContext, TEntity, TSelectedFields>[];
+  beforeUpdate?: EntityMutationTrigger<TFields, TID, TViewerContext, TEntity, TSelectedFields>[];
+  beforeDelete?: EntityMutationTrigger<TFields, TID, TViewerContext, TEntity, TSelectedFields>[];
+  beforeAll?: EntityMutationTrigger<TFields, TID, TViewerContext, TEntity, TSelectedFields>[];
```

Comment:
Thinking about this more, there should only be one set of validators and it should run on create and update. This matches active record https://guides.rubyonrails.org/active_record_validations.html

Because of this, I'd say we can just get rid of `EntityMutationValidatorConfiguration` and just have the mutation validators field be an array: `mutationValidators: EntityMutationTrigger<TFields, TID, TViewerContext, TEntity, TSelectedFields>[];`